lib/routes.php

```diff
@@ -17,7 +17,7 @@
 if ($representation != '') {
   if (!$page_exists) {
     header('HTTP/1.0 404 Not Found');
-    die("Page not found: <a href=".EDITH_URI."/".$page->name.">Create me!</a>");
+    echo "404 not Found: The page does not exist.";
```

Comment:
One issue with this approach is that now the page looks like this:

```
404 not Found: The page does not exist.
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>404</title>
  <meta content="width=device-width" name="viewport" />
  <link rel="shortcut icon" href="/public/icon.png" />
  <link rel="stylesheet" href="/public/html.css" />
</head>
<body>

<p>Edit me!</p>

  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.0/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="public/vendor/jquery.js"><\/script>')</script>
  <script src="/public/html.js"></script>
</body>
</html>
```

Which is not valid HTML and shows both the 404 text and the “Edit me!” text:

<img width="917" alt="Screenshot of a page with “404 not Found: The page does not exist.” and “Edit me!”" src="https://github.com/sunny/edith/assets/132/95afae24-17a1-49db-a726-a602b0e76aa3">

Ideally we would replace the `Edit me!` with the `404 not Found: The page does not exist` text.

One way we could do this would be to set a variable here and use it inside a condition around the page text.

Reply:
Thanks for the feedback! I was able to make progress on this change, where the 'Edit me!' text is now replaced with valid html. I am running to a logical error, however, where now it seems to display the replaced text when it's not meant to. I added comments on commit 8c4a5c1 to explain in more detail. Why might this be happening?

Reply:
So far I built something like this: 
![Screenshot from 2023-09-29 10-22-17](https://github.com/sunny/edith/assets/54324007/d3b99dc4-2280-45ce-8bfc-22395a559f5c)

Then, when pressing ctrl+e it returns to: 
![Screenshot from 2023-09-29 10-24-23](https://github.com/sunny/edith/assets/54324007/999cc5ee-1565-439a-9a07-fa6814bd4e54)

So it seems to work as intended, with the exception of the `Edit me!` text. There may be some way to use a variable & condition to prevent that text from appearing, but my attempts to implement it so far were unsuccessful. While not ideal, I was able to keep the text within HTML.

Reply:
Good progress!

The reason why you still have `Edit me!` is that now the HTML is two pages: the 404 and the regular page at the same time. If you look at the generated source you will see that it prints `<!DOCTYPE html>` twice, which is not valid:

```
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
	<title>404 Not Found</title>
</head>
<body>
	<h1>404 Not Found</h1>

	<p>Error Message: <?php echo "Page not found"; ?></p>
</body>
</html>
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <title>404</title>
  <meta content="width=device-width" name="viewport" />
  <link rel="shortcut icon" href="/public/icon.png" />
  <link rel="stylesheet" href="/public/html.css" />
</head>
<body>

<p>Edit me!</p>

  <script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.0/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="public/vendor/jquery.js"><\/script>')</script>
  <script src="/public/html.js"></script>
</body>
</html>
```

What you could do is add an `exit` after your call to `include` so that you don’t get both pages included at the same time.

This will remove the ctrl-e functionnality but you can add it back by copying the `<script>` tags inside the 404 page.